input/text/src/commonMain/kotlin/majestic/Select.kt

```diff
@@ -0,0 +1,134 @@
+package majestic
+
+import androidx.compose.animation.core.animateFloatAsState
+import androidx.compose.animation.core.tween
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowDropDown
+import androidx.compose.material3.DropdownMenuItem
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.ExposedDropdownMenuBox
+import androidx.compose.material3.ExposedDropdownMenuDefaults
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MenuAnchorType
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.unit.dp
+import majestic.colors.ColorPair
+
+class SelectMicroColors(
+    val border: Color,
+    val placeholder: Color,
+    val text: Color
+)
+
+class SelectColors(
+    val focused: SelectMicroColors = SelectMicroColors(
+        border = Color(0xFF0061FF),
+        placeholder = Color.Black.copy(alpha = 0.4f),
+        text = Color.Black
+    ),
+    val blurred: SelectMicroColors = SelectMicroColors(
+        border = Color.Black.copy(alpha = 0.2f),
+        placeholder = Color.Black.copy(alpha = 0.4f),
+        text = Color.Black
+    ),
+    val disabled: SelectMicroColors = SelectMicroColors(
+        border = Color.Black.copy(alpha = 0.1f),
+        placeholder = Color.Black.copy(alpha = 0.2f),
+        text = Color.Black.copy(alpha = 0.4f)
+    ),
+    val dropdown: ColorPair = ColorPair(
+        background = Color.White,
+        foreground = Color.Black
+    ),
+)
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun <T> Select(
+    options: List<T>,
+    toString: (T) -> String,
```

Comment:
instead of having a `toString: (T)->String`, add `item: @Composable () -> Unit` and use this to render, giving the consumer full controll of what the selector can look like